upgrade/upgradeV12/upgradeV12.ts

```diff
@@ -0,0 +1,332 @@
+    }
+
+    // Validate each signer object
+    for (let i = 0; i < signersToAdd.length; i++) {
+        const signer = signersToAdd[i];
+
+        // Check that signer is an object with required properties
+        if (!signer || typeof signer !== 'object') {
+            throw new Error(`Signer at index ${i} must be an object with addr and url properties`);
+        }
+
+        if (!signer.addr || !signer.url) {
+            throw new Error(`Signer at index ${i} must have both addr and url properties`);
+        }
+
+        // Validate address
+        if (signer.addr === ethers.ZeroAddress) {
+            throw new Error(`Signer at index ${i} cannot have zero address`);
+        }
+        if (!ethers.isAddress(signer.addr)) {
+            throw new Error(`Invalid address format for signer at index ${i}: ${signer.addr}`);
+        }
+
+        // Validate URL
+        if (typeof signer.url !== 'string' || signer.url.trim().length === 0) {
+            throw new Error(`Signer at index ${i} must have a non-empty URL string`);
+        }
+
+        // Check for duplicate signers
+        for (let j = i + 1; j < signersToAdd.length; j++) {
+            if (signersToAdd[j].addr && signer.addr.toLowerCase() === signersToAdd[j].addr.toLowerCase()) {
+                throw new Error(`Duplicate signer address found: ${signer.addr}`);
+            }
+        }
+    }
+
+    // Check threshold constraints
+    const newThreshold = initializeAgglayerGateway.newThreshold;
+    if (newThreshold > signersToAdd.length) {
+        throw new Error(`Threshold (${newThreshold}) cannot be greater than number of signers (${signersToAdd.length})`);
+    }
+    if (signersToAdd.length > 0 && newThreshold === 0) {
+        throw new Error('Threshold cannot be zero when signers are present');
+    }
+    if (signersToAdd.length > 255) { // MAX_AGGCHAIN_SIGNERS = 255
+        throw new Error(`Number of signers (${signersToAdd.length}) exceeds maximum allowed (255)`);
+    }
+
+    logger.info(`✓ Validation passed: ${signersToAdd.length} signers, threshold: ${newThreshold}`);
+
+    const timelockAddress = await proxyAdmin.owner();
+
+    // load timelock
+    const timelockContractFactory = await ethers.getContractFactory('PolygonZkEVMTimelock', deployer);
+
+    // prepare upgrades
+
+    // 1. Upgrade Rollup Manager
+    logger.info('Preparing Rollup Manager upgrade...');
+    const newRollupManagerFactory = await ethers.getContractFactory('PolygonRollupManager', deployer);
+
+    const implRollupManager = await upgrades.prepareUpgrade(rollupManagerAddress, newRollupManagerFactory, {
+        constructorArgs: [globalExitRootV2Address, polAddress, bridgeV2Address, aggLayerGatewayAddress],
+        unsafeAllow: ['constructor'],
+    });
+
+    logger.info('#######################\n');
+    logger.info(`Polygon rollup manager implementation deployed at: ${implRollupManager}`);
+
+    await verifyContractEtherscan(implRollupManager as string, [
```

Comment:
I'm thinking that it is worth it to add verification information in the output json (separate property `verification`) that contains:
- if verified correctky --> OK
- otherwise, include verification parameters (constructor parameters)